pkg/controller/endpointslice/reconciler.go

```diff
@@ -193,9 +193,10 @@ func (r *reconciler) finalize(
 		// chance EndpointSlices with an IP AddressType would otherwise be
 		// updated to IPv4 or IPv6 without this check.
 		if sliceToDelete.AddressType == slice.AddressType {
-			slice.Name = sliceToDelete.Name
+			epSlice := slice.DeepCopy()
+			epSlice.Name = sliceToDelete.Name
```

Comment:
needing to copy here surprised me. I thought everything in the slicesToCreate list was new and not from the cache and therefore free for us to modify

Reply:
Good catch, thanks. Those slices will always be new objects.